apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl

```diff
@@ -267,6 +292,29 @@ fields(reorder) ->
     [
         {order, mk(array(binary()), #{required => true, in => body})}
     ];
+fields(test_transformation) ->
+    [
+        {transformation,
+            mk(
+                hoconsc:ref(emqx_message_transformation_schema, transformation),
+                #{required => true, in => body}
+            )},
+        {message, mk(ref(test_input_message), #{required => true, in => body})}
+    ];
+fields(test_input_message) ->
+    %% See `emqx_message_transformation:eval_context()'.
+    [
+        {client_attrs, mk(map(), #{required => true})},
+        {payload, mk(binary(), #{required => true})},
+        {qos, mk(range(0, 2), #{required => true})},
+        {retain, mk(boolean(), #{required => true})},
+        {topic, mk(binary(), #{required => true})},
+        {user_property,
+            mk(
+                typerefl:alias("map(binary(), binary())", user_property()),
+                #{required => true}
```

Comment:
why `required => true` for all ?

Reply:
It's better to be explicit about the input, I think.

Reply:
Is the user forced to add all fields in order to trigger a dryrun ?

Reply:
Yes.

Reply:
Most of them don't have a good default, I think,